Mathlib/Topology/Algebra/Order/UpperLower.lean

```diff
@@ -59,7 +60,38 @@ instance (priority := 100) OrderedCommGroup.to_hasUpperLowerClosure [OrderedComm
 #align ordered_comm_group.to_has_upper_lower_closure OrderedCommGroup.to_hasUpperLowerClosure
 #align ordered_add_comm_group.to_has_upper_lower_closure OrderedAddCommGroup.to_hasUpperLowerClosure
 
-variable [Preorder α] [HasUpperLowerClosure α] {s : Set α}
+variable [Preorder α]
+
+section OrderClosedTopology
+variable [OrderClosedTopology α] {s : Set α}
+
+@[simp] lemma upperBounds_closure (s : Set α) : upperBounds (closure s : Set α) = upperBounds s :=
+  ext fun a ↦ by simp_rw [mem_upperBounds_iff_subset_Iic, isClosed_Iic.closure_subset_iff]
+#align upper_bounds_closure upperBounds_closure
+
+@[simp] lemma lowerBounds_closure (s : Set α) : lowerBounds (closure s : Set α) = lowerBounds s :=
+  ext fun a ↦ by simp_rw [mem_lowerBounds_iff_subset_Ici, isClosed_Ici.closure_subset_iff]
+#align lower_bounds_closure lowerBounds_closure
+
+@[simp] lemma bddAbove_closure : BddAbove (closure s) ↔ BddAbove s := by
+  simp_rw [BddAbove, upperBounds_closure]
+#align bdd_above_closure bddAbove_closure
+
+@[simp] lemma bddBelow_closure : BddBelow (closure s) ↔ BddBelow s := by
+  simp_rw [BddBelow, lowerBounds_closure]
+#align bdd_below_closure bddBelow_closure
+
+protected alias ⟨BddAbove.of_closure, BddAbove.closure⟩ := bddAbove_closure
+#align bdd_above.of_closure BddAbove.of_closure
+#align bdd_above.closure BddAbove.closure
+
+protected alias ⟨BddBelow.of_closure, BddBelow.closure⟩ := bddBelow_closure
```

Comment:
`BddBelow.of_closure` doesn't need any assumptions on the relation of the topology and the order. Please prove it without assumptions.